src/main/java/org/jitsi/jicofo/jigasi/JigasiDetector.java

```diff
@@ -91,45 +111,121 @@ protected void onInstanceStatusChanged(
     protected void notifyInstanceOffline(Jid jid)
     {}
 
+    /**
+     * Selects the jigasi instance that is less loaded.
+     * @param preferredRegions a list of preferred regions.
+     * @return XMPP address of Jigasi instance or <tt>null</tt> if there are
+     * no Jigasis available currently.
+     */
+    public Jid selectTranscriber(
+        List<Jid> filter, List<String> preferredRegions)
+    {
+        return this.selectJigasi(
+            instances, filter, preferredRegions, localRegion, true);
+    }
+
     /**
      * Selects the jigasi instance that is less loaded.
      *
+     * @param filter a list of <tt>Jid</tt>s to be filtered from the list of
+     * available Jigasi instances. List that we do not want as a result.
+     * @param preferredRegions a list of preferred regions.
      * @return XMPP address of Jigasi instance or <tt>null</tt> if there are
      * no Jigasis available currently.
      */
-    public Jid selectJigasi()
+    public Jid selectJigasi(List<Jid> filter, List<String> preferredRegions)
```

Comment:
We don't care about the order of the preferred regions, right? Can we make it a `Collection` or a `Set` and make sure it's non-null and doesn't contain null elements? 

In the future we will be using more than one bridge in a region, and not every deployment has regions configured (in which case the bridges' regions are null).